src/backend/distributed/planner/multi_join_order.c

```diff
@@ -1423,7 +1456,8 @@ CartesianProduct(JoinOrderNode *currentJoinNode, TableEntry *candidateTable,
 {
 	JoinOrderNode *nextJoinNode = MakeJoinOrderNode(candidateTable, CARTESIAN_PRODUCT,
 													currentJoinNode->partitionColumn,
-													currentJoinNode->partitionMethod);
+													currentJoinNode->partitionMethod,
```

Comment:
I think we should explicitly pass `NULL` as we did for `DualHashPartition`. What do you think? Because we're losing the `anchor` notion anyway.